Objects/funcobject.c

```diff
@@ -327,17 +375,86 @@ func_set_defaults(PyFunctionObject *op, PyObject *value)
     return 0;
 }
 
+static PyObject *
+func_get_defaults(PyFunctionObject *op)
+{
+     if (PyErr_WarnPy3k_WithFix("The attribute func_defaults is not supported in 3.x, ",
+                                "use '__defaults__' instead", 1) < 0)
+        return NULL;
+    return get_defaults(op);
+}
+
+static int
+func_set_defaults(PyFunctionObject *op, PyObject *value)
+{
+     if (PyErr_WarnPy3k_WithFix("The attribute func_defaults is not supported in 3.x, ",
+                                "use '__defaults__' instead", 1) < 0)
+        return -1;
+    return set_defaults(op, value);
+}
+
+static PyObject *
+func_get_closure(PyFunctionObject *op)
+{
+    if (PyErr_WarnPy3k_WithFix("The attribute func_closure is not supported in 3.x, ",
+                               "use '__closure__' instead", 1) < 0)
+        return NULL;
+    if (restricted())
+        return NULL;
+    if (op->func_closure == NULL) {
+        Py_INCREF(Py_None);
+        return Py_None;
+    }
+    Py_INCREF(op->func_closure);
+    return op->func_closure;
+}
+
+static PyObject *
+func_get_doc(PyFunctionObject *op)
+{
+    if (PyErr_WarnPy3k_WithFix("The attribute func_doc is not supported in 3.x, ",
+                               "use '__doc__' instead", 1) < 0)
+        return NULL;
+    if (restricted())
+        return NULL;
+    if (op->func_doc == NULL) {
+        Py_INCREF(Py_None);
+        return Py_None;
+    }
+    Py_INCREF(op->func_doc);
+    return op->func_doc;
+}
+
+static PyObject *
+func_get_globals(PyFunctionObject *op)
+{
+    if (PyErr_WarnPy3k_WithFix("The attribute func_globals is not supported in 3.x, ",
+                               "use '__globals__' instead", 1) < 0)
+        return NULL;
+    if (restricted())
+        return NULL;
+    if (op->func_globals == NULL) {
+        Py_INCREF(Py_None);
```

Comment:
Surprised you have to refcount `None`, which I'd expect to be an eternal singleton. Is it not?

Reply:
Technically not, it seems to segfault without also not too wrong since its done elsewhere.

Reply:
Happy to know if you have any reason to think otherwise

Reply:
I believe refcounting `None` is standard practise.